lib/nats.js

```diff
@@ -824,10 +843,13 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   } else {
     psub = 'PUB ' + subject + SPC + opt_reply + SPC;
   }
+
   this.sendCommand(psub + Buffer.byteLength(msg) + CR_LF + msg + CR_LF);
 
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
+  } else if (this.closed) {
+    this.emit('error', PUB_CONN_CLOSED_ERR);
```

Comment:
So this one should not emit either? So if no callback is provided, then the user would not have an indication of an error? This seems to be forcing users into always providing callbacks which is not efficient and generally an anti-pattern IMO.